src/common/imageio.c

```diff
@@ -1280,6 +1283,11 @@ dt_imageio_retval_t dt_imageio_open(dt_image_t *img,               // non-const
   /* check if file is ldr using magic's */
   if(dt_imageio_is_ldr(filename)) ret = dt_imageio_open_ldr(img, filename, buf);
 
+#ifdef HAVE_LIBJXL
```

Comment:
Perhaps move this into HDR loaders, keep together w/ AVIF/HEIF as "modern" formats?

Reply:
> Perhaps move this into HDR loaders, keep together w/ AVIF/HEIF as "modern" formats?

We can't put the JXL loader in HDR loaders because then it won't be able to tell dt that the image is LDR because dt_imageio_open_hdr will override that flag to DT_IMAGE_HDR.

> Alternatively, we could also later refactor AVIF/HEIF loaders to set the HDR image flags accordingly like this one and take them out of the generic HDR loaders?

Yes, that's exactly how it should be. I'm currently preparing a PR for loaders of "potentially HDR" formats, which will take their calls from dt_imageio_open_hdr and add them to the main queue.

Reply:
Great, let's tackle that separately.